src/java/main/org/apache/zookeeper/server/command/FourLetterCommands.java

```diff
@@ -153,13 +155,33 @@
      */
     public final static int telnetCloseCmd = 0xfff4fffd;
 
-    final static HashMap<Integer, String> cmd2String =
-        new HashMap<Integer, String>();
+    private static final String ZOOKEEPER_4LW_COMMANDS_WHITELIST = "zookeeper.4lw.commands.whitelist";
+
+    final static Map<Integer, String> cmd2String = new HashMap<Integer, String>();
+
+    final static Set<String> whiteListedCommands = new HashSet<String>();
 
     public static Map<Integer, String> getCmdMapView() {
         return Collections.unmodifiableMap(cmd2String);
     }
 
+    // ZOOKEEPER-2693: Only allow white listed commands.
+    public static Set<String> getWhiteListedCmdView() {
+        if (!whiteListedCommands.isEmpty()) {
+            return Collections.unmodifiableSet(whiteListedCommands);
+        }
+
+        String commands = System.getProperty(ZOOKEEPER_4LW_COMMANDS_WHITELIST);
+        if (commands != null) {
+            String[] list = commands.split(",");
+            for (String cmd : list) {
+                whiteListedCommands.add(cmd.trim());
```

Comment:
That is fine - an empty string will not do any damage here but it is a good to have check and the cost is minimum. Will do.